ColorUtility.cpp

```diff
@@ -0,0 +1,49 @@
+#include "ColorUtility.h"
+#include <iomanip>
+namespace TelCoColorCoder
+{
+    extern int numberOfMajorColors;
+    extern int numberOfMinorColors;
+
+    ColorPair GetColorFromPairNumber(int pairNumber) 
+    {
+        int zeroBasedPairNumber = pairNumber - 1;
+        MajorColor majorColor =
+            (MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
+        MinorColor minorColor =
+            (MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
+        return ColorPair(majorColor, minorColor);
+    }
+
+    int GetPairNumberFromColor(MajorColor major, MinorColor minor)
+    {
+        return major * numberOfMinorColors + minor + 1;
+    }
+    
+    // Original
+    void PrintColorManual(std::ostream& out)
```

Comment:
This function is not tested. Perhaps it's not required anymore, since the other one is a superset of this?

Reply:
yes, the other overloaded function is superset of this